bin/GoogleAdsCleanupServices.php

```diff
@@ -0,0 +1,529 @@
+		$this->remove_use_statement( $file, "{$service}ServiceClient" );
+		$this->remove_function( $file, "get{$service}ServiceClient" );
+
+		$this->remove_file( "/{$this->version}/Services/Gapic/{$service}ServiceGapicClient.php" );
+		$this->remove_file( "/{$this->version}/Services/{$service}ServiceGrpcClient.php" );
+		$this->remove_file( "/{$this->version}/Services/{$service}ServiceClient.php" );
+
+		$this->remove_file( "/{$this->version}/Services/Mutate{$service}Result.php" );
+		$this->remove_file( "/{$this->version}/Services/Mutate{$service}sRequest.php" );
+		$this->remove_file( "/{$this->version}/Services/Mutate{$service}sResponse.php" );
+		$this->remove_file( "/{$this->version}/Services/{$service}Operation.php" );
+		$this->remove_file( "/{$this->version}/Services/Get{$service}Request.php" );
+		$this->remove_file( "/{$this->version}/Resources/{$service}.php" );
+
+		$snake_case = strtolower( preg_replace( '/(?<!^)[A-Z]/', '_$0', $service ) );
+		$this->remove_file( "/{$this->version}/Services/resources/{$snake_case}_service_descriptor_config.php" );
+		$this->remove_file( "/{$this->version}/Services/resources/{$snake_case}_service_rest_client_config.php" );
+		$this->remove_file( "/{$this->version}/Services/resources/{$snake_case}_service_client_config.json" );
+
+		// Remove metadata files.
+		$this->src_path = '/metadata/Google/Ads/GoogleAds';
+		$this->remove_file( "/{$this->version}/Services/{$service}Service.php" );
+		$this->remove_file( "/{$this->version}/Resources/{$service}.php" );
+	}
+
+	/**
+	 * Find used enums and remove any unused ones.
+	 */
+	protected function remove_enums() {
+		$library_enums = $this->find_library_enums();
+		$used_enums    = $this->find_used_enums();
+
+		foreach ( array_diff( $library_enums, $used_enums ) as $enum ) {
+			$this->remove_enum( $enum );
+		};
+	}
+
+	/**
+	 * Remove a specific enum.
+	 *
+	 * @param string $enum Enum name.
+	 */
+	protected function remove_enum( string $enum ) {
+		$this->output_text( "Removing enum {$enum}" );
+		$this->src_path = '/src/Google/Ads/GoogleAds';
+
+		$this->remove_file( "/{$this->version}/Enums/{$enum}Enum.php" );
+		$this->remove_file( "/{$this->version}/Enums/{$enum}Enum_{$enum}.php" );
+		$this->remove_file( "/{$this->version}/Enums/{$enum}Enum/{$enum}.php" );
+		$this->remove_directory( "/{$this->version}/Enums/{$enum}Enum" );
+
+		// Remove metadata files.
+		$this->src_path = '/metadata/Google/Ads/GoogleAds';
+		$this->remove_file( "/{$this->version}/Enums/{$enum}.php" );
+	}
+
+	/**
+	 * Find enum names within the library.
+	 *
+	 * @return array
+	 */
+	protected function find_library_enums(): array {
+		$path = "{$this->path}/metadata/Google/Ads/GoogleAds/{$this->version}/Enums";
+
+		$output = glob( "{$path}/*.php" );
+
+		if ( empty( $output ) ) {
+			return [];
+		}
+
+		return array_map(
+			function( $file ) {
+				preg_match( '/(.*)\.php/', basename( $file ), $matches );
+				return $matches[1];
```

Comment:
💅 

`return pathinfo($file, PATHINFO_FILENAME);` also works and is easier to read.